tests/debug-tests/recv.py

```diff
@@ -0,0 +1,223 @@
+import asyncio
+import os
+
+from distributed.comm.utils import from_frames
+from distributed.utils import nbytes
+
+import cloudpickle
+import numpy as np
+import pynvml
+import pytest
+import rmm
+import ucp
+
+# import subprocess
+
+
+pynvml.nvmlInit()
+
+
+cmd = "nvidia-smi nvlink --setcontrol 0bz"  # Get output in bytes
+# subprocess.check_call(cmd, shell=True)
+
+pynvml = pytest.importorskip("pynvml", reason="PYNVML not installed")
+ITERATIONS = 1000
+
+
+def cuda_array(size):
+    # import cupy as cp
+
+    # return numba.cuda.device_array((size,), dtype=np.uint8)
+
+    # return cp.empty(size, dtype=cp.uint8)
+    return rmm.device_array(size, dtype=np.uint8)
```

Comment:
I wonder if we could use a parameterized fixture to try all 3.

ref: https://docs.pytest.org/en/stable/fixture.html#parametrizing-fixtures

Reply:
The debug tests of send/recv are the equivalent of test_send_recv_two_workers.py but separated into two processes.  This was needed during debugging to help isolate where cudaFree/cudaMalloc were causing problems.  these test are useful for debugging will probably not run in CI.  Over time, I suspect we can remove them